src/coreclr/tests/src/Interop/PInvoke/Generics/GenericsTest.NullableB.cs

```diff
@@ -11,24 +11,15 @@ unsafe partial class GenericsNative
     [DllImport(nameof(GenericsNative))]
     public static extern bool? GetNullableB(bool hasValue, bool value);
 
-    [DllImport(nameof(GenericsNative))]
-    public static extern void GetNullableBOut(bool hasValue, bool value, bool?* pValue);
```

Comment:
Rather than deleting this test code, couldn't we just update the test to validate it functions as expected?

Reply:
The scenarios in the GenericsTest that I deleted aren't exactly expected to function "correctly". They're just expected to not throw exceptions since we're assuming that "you know what you're doing". There's no good way to use a pointer to `bool?` from native, but if you want to pass it down you can. The only "correct" behavior that we could validate was that we passed down a pointer, which isn't exactly a useful test to add in addition to the targeted "nonblittable pointer" test.

Reply:
Is this because `bool*` would function as a `uint8_t*`, since no marshalling happens.

However, `ref bool` and `out bool` may function as `uint32_t*` (and be impacted by `In, Out`)?

Reply:
Yep. `bool*` functions as a `uint8_t*` since there's no marshalling,. A `bool?*` would function as a `{ bool; bool; }*` since you'd get the managed layout passed directly down to native code without any marshalling.

Reply:
>  A bool?* would function as a { uint8_t; uint8_t; }*

Note that the exact nullable type layout is not guaranteed anywhere. You cannot tell which one of these `uint8_t` is the Value and which one is the HasValue flag.